fairroot/base/sim/FairVolumeList.cxx

```diff
@@ -58,7 +58,7 @@ FairVolume* FairVolumeList::findObject(TString name)
         obj = static_cast<FairVolume*>(fData->At(i));
         if (obj) {
             if (obj->GetName() == name) {
-                return static_cast<FairVolume*>(obj);
+                return obj;
             }
         }
     }
```

Comment:
How about?

```cpp
FairVolume* FairVolumeList::findObject(TString name)
{
    return fData->FindObject(name.Data());
}
```

see https://root.cern/doc/v630/classTObjArray.html#a9bae82c7ff2a18715a90602e2aedca61

Reply:
Yeah, I was wondering, why this was not used like this. I was rather thinking of replacing TObjArray with a `std::vector<std::unique_ptr<FairVolume>>`. And then this linear search would be needed again.

Or maybe we should create something like this?
```cpp
namespace fairroot::detail
{
template<typename T>
T* FindObject(std::vector<std::unique_ptr<T>>& v, std::string_view v)
{
    auto name_matches = [&](std::unique_ptr<TObject>& p) { return p->GetName() == name; };
    auto it = std::find_if(std::begin(v), std::end(v), name_matches);
    if (it == std::end(v))
        return nullptr;
    return it->get();
}
}
```

Reply:
Hm, why even hardcode `vector<unique_ptr<T>>`?

Reply:
And `FindObject` would be a terrible name. More like `FindByName`?

Reply:
But thinking about what `FairVolumeList` wants to be, isn't it not just a set?

Reply:
Looking at where `FairVolumeList::findObject()` is called, I find [`FairModule::AddSensitiveVolume()`](https://github.com/FairRootGroup/FairRoot/blame/00249b0fc465bace608c4a3c5a9291e015c86895/base/sim/FairModule.cxx#L295-L311) which searches the "list" twice (linearly) since [12 years](https://github.com/FairRootGroup/FairRoot/commit/386912ebcc7877927980cbbf7ef46a32dddc3e4b).

We should definitely rename this class to reflect it is a "Set" ;)

Reply:
https://github.com/FairRootGroup/FairRoot/issues/1495

Reply:
> Hm, why even hardcode `vector<unique_ptr<T>>`?

Because of the `it->get()` at the end.

Reply:
> #1495

Now that we have a dedicated bug to look at all of this: What about we merge this PR just "as is"?